### DISCUSSION TITLE
Partial inline widget support

### DISCUSSION THREAD

**Discussion by matthew-carroll** [Flutter-Bounty-Hunters/super_editor#468] [Milestones]
Support a subset of inline widget use-cases:

- Auto-replace keywords, like `:fire:`, with a small bitmap (non-emoji). The bitmap can appear anywhere in the text, and therefore it needs to play nice with text selection.
- Show progress at the end of a text blob, e.g., "✅ 0/2". This widget should not be selectable at all.

The reason that we're supporting partial inline widgets, instead of full-featured inline widgets, is because Flutter's text layout algorithm can't break a widget across multiple lines, which is required for use-cases like inline code.

### Funding
Superlist, Turtle

**Comment by bk-one** [comment]
This is a requirement for Superlist and I'd like to see this latest directly after 0.3.
